server/src/main/java/com/project/yozmcafe/service/CafeService.java

```diff
@@ -66,4 +69,18 @@ public CafeResponse getCafeById(final long cafeId) {
 
         return CafeResponse.fromUnLoggedInUser(foundCafe);
     }
+
+    public List<CafeSearchResponse> getCafesByKeyWord(final String cafeNameKeyWord, final String menuKeyWord, final String addressKeyWord) {
+        List<Cafe> cafes;
+        if (StringUtils.isBlank(menuKeyWord)) {
+            cafes = cafeRepository.findAllBy(cafeNameKeyWord, addressKeyWord);
+        }
+        else {
+            cafes = cafeRepository.findAllBy(cafeNameKeyWord, menuKeyWord, addressKeyWord);
+        }
+
+        return cafes.stream()
+                .map(CafeSearchResponse::from)
+                .toList();
+    }
```

Comment:
```suggestion
    public List<CafeSearchResponse> getCafesByKeyWord(final String cafeNameKeyWord, final String menuKeyWord,
                                                      final String addressKeyWord) {
        final List<Cafe> cafes = asdf(cafeNameKeyWord, menuKeyWord, addressKeyWord).get();

        return cafes.stream()
                .map(CafeSearchResponse::from)
                .toList();
    }

    private Supplier<List<Cafe>> asdf(final String cafeNameKeyWord, final String menuKeyWord,
                                      final String addressKeyWord) {
        if (menuKeyWord.isBlank()) {
            return () -> cafeRepository.findAllBy(cafeNameKeyWord, addressKeyWord);
        }
        return () -> cafeRepository.findAllBy(cafeNameKeyWord, menuKeyWord, addressKeyWord);
    }
```

저도 재할당은 좀 보기싫어서 좋은 방법 없을까 하다가.
이런건 어떤가요?
펑셔널인터페이스 맛도리~

Reply:
폴로 천재!